.github/workflows/ci.yml

```diff
@@ -0,0 +1,290 @@
+          npm run mmjstool -- i18n clean-empty --webapp-dir . --mobile-dir /tmp/fake-mobile-dir --check
+          npm run mmjstool -- i18n check-empty-src --webapp-dir . --mobile-dir /tmp/fake-mobile-dir
+          rm -rf tmp
+  check-type:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: ci/setup-node
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        id: setup_node
+        with:
+          node-version-file: ".nvmrc"
+          cache: npm
+      - uses: actions/cache@2b8105bdae4b746666ee225970c4a281bbd9d51f # v3.2.4
+        id: npm-cache
+        with:
+          path: |
+            '**/node_modules'
+            'e2e/playwright/node_modules'
+            'e2e/cypress/node_modules'
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('e2e/cypress/package-lock.json') }}-${{ hashFiles('e2e/playwright/package-lock.json') }}
+      - name: ci/get-node-modules
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: |
+          make node_modules
+          make e2e/playwright/node_modules
+      - name: ci/lint
+        run: |
+          npm run check-types
+  tests:
+    runs-on: ubuntu-22.04
+    needs: [check-type, check-i18n, check-lint]
+    permissions:
+      checks: write
+      pull-requests: write
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: ci/setup-node
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        id: setup_node
+        with:
+          node-version-file: ".nvmrc"
+          cache: npm
+      - uses: actions/cache@2b8105bdae4b746666ee225970c4a281bbd9d51f # v3.2.4
+        id: npm-cache
+        with:
+          path: |
+            '**/node_modules'
+            'e2e/playwright/node_modules'
+            'e2e/cypress/node_modules'
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('e2e/cypress/package-lock.json') }}-${{ hashFiles('e2e/playwright/package-lock.json') }}
+      - name: ci/get-node-modules
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: |
+          make node_modules
+          make e2e/playwright/node_modules
+      - name: ci/detect-cpu-core
+        id: cpu-cores
+        uses: SimenB/github-actions-cpu-cores@c508d404ab007faae80a014072fd8c0e17792118 # v1.1.0
+      - name: ci/test
+        run: |
+          npm run test-ci -- --maxWorkers=${{ steps.cpu-cores.outputs.count }}
+      - name: ci/publish-test-results
+        uses: EnricoMi/publish-unit-test-result-action@a3caf02865c0604ad3dc1ecfcc5cdec9c41b7936  # v2.3.0
+        if: always()
+        with:
+          junit_files: "build/**/*.xml"
+          comment_mode: always
+          compare_to_earlier_commit: false
+  build:
```

Comment:
I know we are just transferring from CIrcleCI but I would prefer to build and push to S3 after all the tests have passed . 
@hmhealey for feedback . 
If we have performance issues we can still bump the resources on our Github Runners to speed tests . 
cc: @phoinixgrr

Reply:
I agree. 
@hmhealey Let us know your thoughts on this recommendation. 

If Harrison confirms, To keep the scope of this PR limited, Let's tackle this in a Separate PR, after this gets merged

Reply:
I need to double check my post in the Platform-Release Collaboration channel about all the artifacts in S3, but that should be fine. The only thing we might lose out on by doing that though is the ability to run test servers for PRs that have in-progress unit tests since that's the main thing that uses webapp-generated builds from S3

Reply:
There is no additional cost involved running as is, since we manage to run the pipeline in the Free Tier GitHub Runners for Open Source Projects.  

I suggest merging this as is, and revisit when @hmhealey has some context to share. To keep the scope of this PR limited.